### PR TITLE
bpo-37640: Fix telnetlib crash in Python3 while receiving un-printable characters from server

### PR DESCRIPTION
**Pull request by peternguyen93**
https://bugs.python.org/issue37640

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by tirkarthi** [comment]
Please change the PR title to be of the format `bpo-NNNN: Description of the fix` where NNNN is the issue number to make sure the PR is linked to the issue. This PR would require tests and a NEWS entry. Given that the fix is applied on master branch I am not sure if the code to detect to be above python 3.6 is needed.

**Comment by ericvsmith** [comment]
@tirkarthi: This code doesn't need a version test. The stdlib always only runs on the version it's shipped with. Agreed about needing tests and NEWS.

**Comment by peternguyen93** [comment]
Hi everyone, this issue should be fixing because a lot of CTF tools are based on it if it upgrades to python 3

File: Lib/telnetlib.py [hunk 1]
```diff
@@ -38,6 +38,9 @@
 import selectors
 from time import monotonic as _time
 
+# get current python version
+is_py_3 = sys.version_info >= (3, 6)
+
```
Comment (ericvsmith) [inline Lib/telnetlib.py:43]: Please remove the version tests. This code only needs to run on Python 3.

**Review by ericvsmith — CHANGES_REQUESTED** [review]
You need to add tests and a Misc/NEWS entry (using blurb).

File: Lib/telnetlib.py [hunk 2]
```diff
@@ -552,7 +555,11 @@ def interact(self):
                             print('*** Connection closed by remote host ***')
                             return
                         if text:
-                            sys.stdout.write(text.decode('ascii'))
+                            if is_py_3:
```
Comment (matrixise) [inline Lib/telnetlib.py:558]: just 
```python
sys.stdout.buffer.write(text)
```

File: Lib/telnetlib.py [hunk 3]
```diff
@@ -579,7 +586,11 @@ def listener(self):
                 print('*** Connection closed by remote host ***')
                 return
             if data:
-                sys.stdout.write(data.decode('ascii'))
+                if is_py_3:
```
Comment (matrixise) [inline Lib/telnetlib.py:589]: ditto

**Review by matrixise — CHANGES_REQUESTED** [review]
Thank you for your contribution but
1. Sign the CLA
2. Remove the test and simplify the code.

Thank you

**Comment by csabella** [comment]
@peternguyen93, just another reminder to sign the CLA and address the code review.  Without that, we'll have to close your PR.  Thanks!

**Comment by csabella** [comment]
I'm going to close this pull request as the author has not signed the CLA after more than one request to do so.  Once the CLA is signed, this can be reopened.  Thank you.

**Pull request closed without merging**